### PR TITLE
feat: add support for regexi operator

### PR DESCRIPTION
**Pull request by vazarkevych**
This PR adds support for case-insensitive operators in the condition evaluator.

1. Adds support for case-insensitive regex operators:
- `$regexi`

- `$notRegexi`

- `$notRegex`

2. Adds case-insensitive versions of membership operators:
- `$ini` — case-insensitive version of `$in`

- `$nini` — case-insensitive version of `$nin`

- `$alli` — case-insensitive version of `$all`.

### CODE REVIEW

File: lib/src/Evaluator/condition_evaluator.dart [hunk 1]
```diff
@@ -302,34 +314,25 @@ class GBConditionEvaluator {
         case '\$in':
           return isIn(attributeValue, conditionValue);
```
Comment (madhuchavva) [inline lib/src/Evaluator/condition_evaluator.dart:316]: The new operators are useful, and the tests cover the happy paths well. Since this adds public condition syntax support (`$regexi`, `$notRegex`, `$notRegexi`, `$ini`, `$nini`, `$alli`), can we add README/changelog or docs coverage for the supported operators and expected behavior? It would also be good to confirm these semantics match the other SDKs.

File: lib/src/Evaluator/condition_evaluator.dart [hunk 2]
```diff
@@ -422,10 +452,73 @@ class GBConditionEvaluator {
     return false;
   }
 
-  bool isIn(dynamic actualValue, List<dynamic> conditionValue) {
+  bool isIn(dynamic actualValue, List<dynamic> conditionValue,
```
Comment (madhuchavva) [inline lib/src/Evaluator/condition_evaluator.dart:455]: Can we add edge-case tests for mixed non-string values in the case-insensitive membership operators? The implementation lowercases only strings and compares other values as-is, which may be fine, but it should be intentional and consistent with other SDKs.

**Review by madhuchavva — COMMENTED** [review]
Inline review comments on new condition operator docs and edge-case semantics.

**Review by madhuchavva — CHANGES_REQUESTED** [review]
Requesting changes based on the inline review comments. The notes call out functional gaps or compatibility/documentation risks that should be addressed before merge.

File: lib/src/Evaluator/condition_evaluator.dart (hunk 1 above)
Comment (vazarkevych) [reply lib/src/Evaluator/condition_evaluator.dart:316]: Added a Condition Operators section in the README listing all supported operators with descriptions and usage examples. Confirmed semantics against the shared GrowthBook spec (specVersion: 0.7.2) — all six operators are defined there and our implementation matches the expected behavior.

File: lib/src/Evaluator/condition_evaluator.dart (hunk 2 above)
Comment (vazarkevych) [reply lib/src/Evaluator/condition_evaluator.dart:455]: Added explicit tests for non-string values in `$ini`, `$nini`, and `$alli`: numbers, null, booleans, and mixed-type lists. The behavior — lowercase strings, compare everything else as-is — is intentional and consistent with the spec, which only defines string-based cases. The new tests make this contract explicit.

**Comment by vazarkevych** [comment]
Closing in favor of #163.

  The original branch (`feature/case-insensitive-regex`) was created in January 2025 and never rebased. Merging it would have reverted
  multiple already-merged PRs:

  - #143 (304 Not Modified handling)
  - #145 (corrupt cache guard, removeContent method)
  - #147 (Uri-based URL construction)
  - #150 (customFields on experiments)
  - #152 (forced variations double format)
  - Atomic file rename in caching_manager
  - ETag cache mechanism

  It also introduced a broken syntax block in network.dart (304 catch indentation), missing `ttlSeconds` param in growth_book_sdk.dart,
  and a `caseFole`/`caseFold` typo.

  The new PR contains only the operator additions cleanly against current main, addresses both review concerns from @madhuchavva (README +
   non-string edge-cases), and fixes the `caseFole` typo.
